src/SearchParameters/shortener.js

```diff
@@ -81,5 +88,22 @@ module.exports = {
    */
   encode: function(key) {
     return keys2Short[key];
+  },
+  /**
+   * Set an attribute / short attribute pair
+   * @param {string} key the attribute
+   * @param {string} shortKey the shorten attribute
+   */
+  set: function(key, shortKey) {
+    var oldShort = keys2Short[key];
+    var idx = ENCODED_PARAMETERS.indexOf(oldShort);
+    if (idx === -1) {
+      ENCODED_PARAMETERS.push(shortKey);
+    } else {
+      ENCODED_PARAMETERS[idx] = shortKey;
+      delete short2Keys[oldShort];
```

Comment:
Isn't `oldshort` a value? Also you can't/shouldn't delete an array entry with delete. You should use splice but I don't think it is necessary, since you are replacing the value.